pkg/command/root.go

```diff
@@ -440,6 +445,10 @@ var k8sCmd = &cobra.Command{
 	Use:     "kubernetes",
 	Short:   "Commands that interact with a Kubernetes endpoint",
 	Aliases: []string{"k8s"},
+	// We are moving away from the 'kubernetes' target.
+	// All commands under this target are accessible as sub-commands of the root command.
+	// For backwards compatibility, we are keeping the target but are hiding it.
+	Deprecated: `you should invoke its sub-commands directly without the "kubernetes" prefix.`,
```

Comment:
> Note that to show the deprecation message for any command invoked using tanzu kubernetes ... I had to put the deprecation message on the plugin command itself. This has the (possibly good) side-effect of hiding all kubernetes plugin commands in the help of tanzu kubernetes -h and also turning off shell-completion in tanzu kubernetes <TAB>.

While I understand why the side-effect happens, it was still somewhat surprising as a user. I think we are considering at least one situation where a command group (apps, v1) _should_ be visible , so perhaps we should still explore just making the output the same as before.

Also:
The side-effect does not appear to apply to remapped commands

```
> tz kubernetes
Command "kubernetes" is deprecated, you should invoke its sub-commands directly without the "kubernetes" prefix.
Commands that interact with a Kubernetes endpoint

Usage:
  tanzu kubernetes [command]

Aliases:
  kubernetes, k8s

Available command groups:

  Manage
    cluster                 remapped commands

Flags:
  -h, --help   help for kubernetes

Use "tanzu kubernetes [command] --help" for more information about a command.

> tanzu kubernetes
Commands that interact with a Kubernetes endpoint

Usage:
  tanzu kubernetes [command]

Aliases:
  kubernetes, k8s

Available command groups:

  Manage
    cluster                 deeper commands
    sample                  sample plugin to test command mapping
  Run
    feature                 Operate on features and featuregates

Flags:
  -h, --help   help for kubernetes

Use "tanzu kubernetes [command] --help" for more information about a command.
```

Reply:
Let me look into that